txnkv/txnsnapshot/snapshot.go

```diff
@@ -645,37 +639,23 @@ func (s *KVSnapshot) Get(ctx context.Context, k []byte) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
-
+	// Update the cache.
+	s.UpdateSnapshotCache([][]byte{k}, map[string][]byte{string(k): val})
```

Comment:
Would the `string(k)` invovle an extra allocation in current golang?

Reply:
Maybe we need to use [unsafe cast](https://github.com/pingcap/tidb/blob/69d7770335a7611d05bb2f3a0888c79afb03e92a/pkg/util/hack/hack.go#L29-L34) like tidb.